From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Werner <werner@armbian.com>
Date: Tue, 9 Dec 2025 17:19:52 +0000
Subject: Patching kernel sunxi files
 arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
 arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi

Origin: Subject: arm64:dts: sun50i-h616-x96-mate T95 eth & sd card hack

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
Signed-off-by: Werner <werner@armbian.com>
---
 arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts | 20 +++++++
 arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi         | 26 ++++++++++
 2 files changed, 46 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts b/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
index 978de97f081f..61a621676848 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
@@ -14,10 +14,11 @@
 / {
 	model = "X96 Mate";
 	compatible = "hechuang,x96-mate", "allwinner,sun50i-h616";
 
 	aliases {
+		ethernet0 = &emac1;
 		serial0 = &uart0;
 	};
 
 	chosen {
 		stdout-path = "serial0:115200n8";
@@ -68,10 +69,29 @@ &ehci2 {
 &gpu {
 	mali-supply = <&reg_dcdcc>;
 	status = "okay";
 };
 
+&emac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_aldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
+	status = "okay";
+};
+
+&mdio1 {
+	rmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+	};
+};
+
+
 &hdmi {
 	hvcc-supply = <&reg_bldo1>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
index 79ef11ad3282..8ddfb43831a4 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -415,10 +415,17 @@ ext_rgmii_pins: rgmii-pins {
 				       /* "PI16" Managed by mdio */
 				function = "emac0";
 				drive-strength = <40>;
 			};
 
+			rmii_pins: rmii-pins {
+				pins = "PA0", "PA1", "PA2", "PA3", "PA4",
+				       "PA5", "PA6", "PA7", "PA8", "PA9";
+				function = "emac1";
+				drive-strength = <40>;
+			};
+
 			i2c0_pins: i2c0-pins {
 				pins = "PI5", "PI6";
 				function = "i2c0";
 			};
 
@@ -978,10 +985,29 @@ mdio0: mdio {
 				#address-cells = <1>;
 				#size-cells = <0>;
 			};
 		};
 
+		emac1: ethernet@5030000 {
+			compatible = "allwinner,sun50i-h616-emac";
+			syscon = <&syscon 1>;
+			reg = <0x05030000 0x10000>;
+			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "macirq";
+			resets = <&ccu RST_BUS_EMAC1>;
+			reset-names = "stmmaceth";
+			clocks = <&ccu CLK_BUS_EMAC1>;
+			clock-names = "stmmaceth";
+			status = "disabled";
+
+			mdio1: mdio {
+				compatible = "snps,dwmac-mdio";
+				#address-cells = <1>;
+				#size-cells = <0>;
+			};
+		};
+
 		gpadc: adc@5070000 {
 			compatible = "allwinner,sun50i-h616-gpadc",
 				     "allwinner,sun20i-d1-gpadc";
 			reg = <0x05070000 0x400>;
 			clocks = <&ccu CLK_BUS_GPADC>;
-- 
Armbian